packages/grid/x-data-grid/src/components/base/GridErrorHandler.tsx

```diff
@@ -1,35 +1,30 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import { useGridPrivateApiContext } from '../../hooks/utils/useGridPrivateApiContext';
-import { useGridLogger } from '../../hooks/utils/useGridLogger';
 import { GridMainContainer } from '../containers/GridMainContainer';
-import { ErrorBoundary } from '../ErrorBoundary';
 import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
 
 function GridErrorHandler(props: { children: React.ReactNode }) {
   const { children } = props;
   const apiRef = useGridPrivateApiContext();
-  const logger = useGridLogger(apiRef, 'GridErrorHandler');
   const rootProps = useGridRootProps();
   const errorState = apiRef.current.state.error;
+  const hasError = !!errorState;
 
   return (
-    <ErrorBoundary
-      hasError={errorState != null}
-      api={apiRef}
-      logger={logger}
-      render={(errorProps) => (
+    <React.Fragment>
+      {hasError ? (
         <GridMainContainer>
           <rootProps.components.ErrorOverlay
-            {...errorProps}
-            {...errorState}
+            hasError={hasError}
```

Comment:
```suggestion
            hasError
```

Or just keep the `!!errorState` in its place.

Reply:
I've removed the `hasError` prop completely, since it's not used and the error overlay won't be rendered if there's no error